src/Microsoft.Identity.Web/TokenAcquisition.cs

```diff
@@ -170,11 +171,24 @@ internal partial class TokenAcquisition : ITokenAcquisitionInternal
 
                 context.HandleCodeRedemption(null, result.IdToken);
             }
+            catch (MsalServiceException exMsal) when (IsInvalidClientCertificateError(exMsal))
+            {
+                DefaultCertificateLoader.ResetCertificates(_microsoftIdentityOptions.ClientCertificates);
```

Comment:
I think I get your design here. You reset the cert and the _app, and you force the recreation of the app. Makes sense. 

As a small observation - I noticed that you added quite a few static methods on DefaultCertificateLoader which just pass in the list of `ClientCertificates`. This indicates that you are missing an object, smth like 

```
ICertManager
+ IEnumerable<ClientCertificates>
+ ResetCertificates() 
+ LoadFirstCertificate() 
+ LoadAllCertificates()
```

Reply:
yes. @bgavrilMS : let's revisit this when we implement it in MSAL?